awscli/compat.py

```diff
@@ -87,6 +114,8 @@ def compat_open(filename, mode='r', encoding=None):
 
     raw_input = raw_input
 
+    binary_stdin = sys.stdin
```

Comment:
@JordonPhillips just to clarify here, my original comment was about `binary_stdin`.

> Can we settle on adding a big todo note on combining/unifying these? 

@kyleknap as I mentioned in my first comment, I'm not suggesting using `BinaryStdout` right now.  I'm saying, if we're taking the effort to move things over to `compat.py` in an effort to clean things up, we should at least have a consistent API.

Having a consistent API is independent of how the API is implemented.  We just shouldn't have two APIs for doing similar things.  We have one way to do binary input/output period.